setup.py

```diff
@@ -220,23 +263,37 @@ def find_module_file(module, dirlist):
         log.info("WARNING: multiple copies of %s found", module)
     return os.path.join(list[0], module)
 
+
 class PyBuildExt(build_ext):
 
     def __init__(self, dist):
         build_ext.__init__(self, dist)
         self.failed = []
         self.failed_on_import = []
+        self.missing = []
+        self.srcdir = None
+        self.config_h_vars = None
+        self.inc_dirs = None
+        self.lib_dirs = None
         if '-j' in os.environ.get('MAKEFLAGS', ''):
             self.parallel = True
 
+    def add_ext(self, ext):
+        self.extensions.append(ext)
+
     def build_extensions(self):
+        self.srcdir = sysconfig.get_config_var('srcdir')
```

Comment:
maybe you could use pathlib.Path ?

Reply:
setup.py is used to build Python. When setup.py is called, most Python modules are not available. I prefer to use the bare minimum from the stdlib. For example, "import subprocess" fails with "ModuleNotFoundError: No module named _posixsubprocess". That's why os.system() is used.

Moreover, I prefer to limit changes in this PR.